openquake/server/db/actions.py

```diff
@@ -187,170 +176,140 @@ def print_outputs_summary(outputs, full=True):
+    :returns: None if everything went fine or an error message
+    """
+    dependent = db(
+        'SELECT id FROM job WHERE hazard_calculation_id=?x', job_id)
+    if dependent:
+        return ('Cannot delete calculation %d: there are calculations '
+                'dependent from it: %s' % job_id, [j.id for j in dependent])
+    deleted = db('DELETE FROM job WHERE id=?x', job_id).rowcount
+    if not deleted:
+        return ('Cannot delete calculation %d: ID does not exist' % job_id)
+    deleted = db('DELETE FROM job WHERE id=?x AND user_name=?x',
+                 job_id, user).rowcount
+    if not deleted:
+        return ('Cannot delete calculation %d: belongs to a different user'
+                % job_id)
 
 
-def log(job_id, timestamp, level, process, message):
+def log(db, job_id, timestamp, level, process, message):
     """
     Write a log record in the database
     """
-    db.connection.cursor().execute(
-        'INSERT INTO log (job_id, timestamp, level, process, message) VALUES'
-        '(%s, %s, %s, %s, %s)', (job_id, timestamp, level, process, message))
+    db('INSERT INTO log (job_id, timestamp, level, process, message) '
+       'VALUES (?X)', (job_id, timestamp, level, process, message))
 
 
-def get_log(job_id):
+def get_log(db, job_id):
     """
     Extract the logs as a big string
     """
-    logs = models.Log.objects.filter(job=job_id).order_by('id')
+    logs = db('SELECT * FROM log WHERE job_id=?x ORDER BY id', job_id)
     for log in logs:
         time = str(log.timestamp)[:-4]  # strip decimals
         yield '[%s #%d %s] %s' % (time, job_id, log.level, log.message)
 
 
-def get_output(output_id):
+def get_output(db, output_id):
     """
     :param output_id: ID of an Output object
     :returns: (ds_key, calc_id, dirname)
     """
-    out = models.get(models.Output, pk=output_id)
-    return out.ds_key, out.oq_job.id, os.path.dirname(out.oq_job.ds_calc_dir)
+    out = db('SELECT output.*, ds_calc_dir FROM output, job '
+             'WHERE oq_job_id=job.id AND output.id=?x', output_id, one=True)
+    return out.ds_key, out.oq_job_id, os.path.dirname(out.ds_calc_dir)
 
 
-def save_performance(job_id, records):
+def save_performance(db, job_id, records):
     """
     Save in the database the performance information about the given job
     """
-    for rec in records:
-        models.Performance.objects.create(
-            job_id=job_id, operation=rec['operation'],
-            time_sec=rec['time_sec'], memory_mb=rec['memory_mb'],
-            counts=rec['counts'])
+    rows = [(job_id, rec['operation'], rec['time_sec'], rec['memory_mb'],
+             rec['counts']) for rec in records]
+    db.insert('performance',
+              'job_id operation time_sec memory_mb counts'.split(), rows)
 
 
 # used in make_report
-def fetch(templ, *args):
+def fetch(db, templ, *args):
     """
-    Run queries directly on the database. Return header + rows
+    Run queries directly on the database
```

Comment:
Since this is a bit cryptic, it would be good to add the description of parameters to the docstring